Assets/Scripts/Environment Actions/ColorManager.cs

```
@@ -83,19 +92,23 @@ private void SetNewPair(Scene scene, LoadSceneMode mode)
                 WinScreenController.Instance.WinningPlayer
                 .GetComponentInChildren<SkinnedMeshRenderer>().material;
 
-            CurrentColorPair = _colorPairs.First(p =>
-                p.ComplementaryPlayerColor != null &&
-                TrimObjectName(p.ComplementaryPlayerColor.name) == TrimObjectName(winningPlayerMaterial.name)
-            );
+            if (_playerColorMappings == null)
+            {
+                InitializePlayerColorMappings();
+            }
+            if (_playerColorMappings.ContainsKey(winningPlayerMaterial.color))
+            {
+                CurrentColorPair = _playerColorMappings[winningPlayerMaterial.color];
+            }
         }
     }
 
-    private string TrimObjectName(string name)
+    private void InitializePlayerColorMappings()
     {
-        if (name.Split('(').Length > 0)
+        _playerColorMappings = new Dictionary<Color, ColorPair>();
+        for (int i = 0; i < _playerMaterials.Count; i++)
```

we should throw an error if the playerMaterials count does not equal the correspondingPlayerColorPairs count.

---

will do